### ISSUE TITLE
Contracts have one transaction count instead of zero

### ISSUE THREAD

**Issue by iFA88** [paritytech/parity-ethereum#9902]
Greetings,

Suddenly I have discovered, that during contract creation the contract address nonce increases.
I'm using `Parity-Ethereum/v2.0.9-stable-09f7757-20181028/x86_64-linux-gnu/rustc1.30.0`, and I have tested on 3 blockchain databases.

```
.eth_getTransactionCount("0xbd67ac342ed6e45a2bc947a06323e0ae8d8113cb", hex(6103999))
"0x0"
.eth_getTransactionCount("0xbd67ac342ed6e45a2bc947a06323e0ae8d8113cb", hex(6104000))
"0x1"
```
I use `eth_getTransactionCount` because it returns the transaction number SENT from that address [
https://wiki.parity.io/JSONRPC-eth-module#eth_gettransactioncount ] . A contract should have zero nonce because they can not send any transaction.

**Comment by tomusdrw** [comment]
That's expected. This RPC has historically always returned the state `nonce` of the account. The same goes for `go-ethereum` implementation: https://github.com/ethereum/go-ethereum/blob/master/internal/ethapi/api.go#L1081

I'd say it's more a documentation issue than implementation. I'd rephrase the method description as:
> Returns the current state nonce of an account. Returned value can be used as the `nonce` of next transaction containing this address' signature. For regular accounts it also describes how many transactions have been already sent, and for contracts how many other contracts this one created.

**Comment by iFA88** [comment]
Thank you!

**Comment by iFA88** [comment]
Sorry, but this is also valid for Ethereum classic? Because on the classic chain after contract creation the nonce is 0, (on the foundation is 1) and if they creates internal a contract then is 1 (on the foundation is 2).
